qiskit_nature/transformers/particle_hole_transformer.py

```diff
@@ -28,5 +45,2102 @@ def transform(self, q_molecule: QMolecule) -> QMolecule:
+                    temp_sign_h2 = 1 * sign_no_term
+
+                    ind_new_h2 = [ind_no_term[i_i], ind_no_term[l_l],
+                                  ind_no_term[j_j], ind_no_term[k_k]]
+                    ind_old_h2 = [ind_ini_term[0], ind_ini_term[1],
+                                  ind_ini_term[2], ind_ini_term[3]]
+                    ind_old_h2 = self.last_two_indices_swap(ind_old_h2)
+
+                    h2_new[ind_new_h2[0]][ind_new_h2[1]][
+                        ind_new_h2[2]][ind_new_h2[3]] \
+                        += 0.5 * temp_sign_h2 * \
+                           h2_old[ind_old_h2[0]][ind_old_h2[1]][ind_old_h2[2]][ind_old_h2[3]]
+
+                elif mapping_no_term == ['a', 'adag', 'adag', 'a']:
+                    temp_sign_h2 = 1 * sign_no_term
+
+                    ind_new_h2 = [ind_no_term[j_j], ind_no_term[k_k],
+                                  ind_no_term[i_i], ind_no_term[l_l]]
+                    ind_old_h2 = [ind_ini_term[0], ind_ini_term[1],
+                                  ind_ini_term[2], ind_ini_term[3]]
+                    ind_old_h2 = self.last_two_indices_swap(ind_old_h2)
+
+                    h2_new[ind_new_h2[0]][ind_new_h2[1]][
+                        ind_new_h2[2]][ind_new_h2[3]] \
+                        += 0.5 * temp_sign_h2 * \
+                           h2_old[ind_old_h2[0]][ind_old_h2[1]][ind_old_h2[2]][ind_old_h2[3]]
+
+                elif mapping_no_term == ['a', 'adag', 'a', 'adag']:
+                    temp_sign_h2 = -1 * sign_no_term
+
+                    ind_new_h2 = [ind_no_term[j_j], ind_no_term[l_l],
+                                  ind_no_term[i_i], ind_no_term[k_k]]
+                    ind_old_h2 = [ind_ini_term[0], ind_ini_term[1],
+                                  ind_ini_term[2], ind_ini_term[3]]
+                    ind_old_h2 = self.last_two_indices_swap(ind_old_h2)
+
+                    h2_new[ind_new_h2[0]][ind_new_h2[1]][
+                        ind_new_h2[2]][ind_new_h2[3]] \
+                        += 0.5 * temp_sign_h2 * \
+                           h2_old[ind_old_h2[0]][ind_old_h2[1]][ind_old_h2[2]][ind_old_h2[3]]
+
+                elif mapping_no_term == ['a', 'a', 'adag', 'adag']:
+                    temp_sign_h2 = 1 * sign_no_term
+
+                    ind_new_h2 = [ind_no_term[k_k], ind_no_term[l_l],
+                                  ind_no_term[i_i], ind_no_term[j_j]]
+                    ind_old_h2 = [ind_ini_term[0], ind_ini_term[1],
+                                  ind_ini_term[2], ind_ini_term[3]]
+                    ind_old_h2 = self.last_two_indices_swap(ind_old_h2)
+
+                    h2_new[ind_new_h2[0]][ind_new_h2[1]][
+                        ind_new_h2[2]][ind_new_h2[3]] \
+                        += 0.5 * temp_sign_h2 * \
+                           h2_old[ind_old_h2[0]][ind_old_h2[1]][ind_old_h2[2]][ind_old_h2[3]]
+
+                else:
+                    print('ERROR 1')
```

Comment:
And can we improve this to raise an error, if that is what is needed (or log a warning or whatever) not print.

Reply:
this is fixed in Max commits